boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,217 @@
+
+        return dict(
+            org=organization, title=organization.title,
+            discount_policies=discount_policies,
+            total_pages=total_pages,
+            paginated=(total_policies > results_per_page),
+            current_page=page
+        )
+    else:
+        return dict(
+            org=organization, title=organization.title
+        )
+
+
+@app.route('/admin/o/<org>/discount_policy/new', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_add_discount_policy(organization):
+    discount_policy_form = DiscountForm()
+    discount_policy_form.items.query = Item.query.all()
+    if not discount_policy_form.validate_on_submit():
+        return make_response(jsonify(status='error', error='invalid_details', error_description=discount_policy_form.errors), 400)
+    discount_policy = DiscountPolicy(title=discount_policy_form.title.data, organization=organization)
+    discount_policy_form.populate_obj(discount_policy)
+    items = request.form.getlist('items')
+    for item_id in items:
+        item = Item.query.get(item_id)
+        discount_policy.items.append(item)
+    if discount_policy_form.discount_type.data:
+        discount_policy.secret = buid()
+    else:
+        discount_policy_form.discount_code_base.data = None
+    db.session.add(discount_policy)
+    db.session.commit()
+    if discount_policy_form.is_price_based.data:
+        item = Item.query.get(items[0])
+        discount_price = Price(discount_policy=discount_policy, title=discount_policy_form.price_title.data, start_at=discount_policy_form.start_at.data, end_at=discount_policy_form.end_at.data, amount=discount_policy_form.amount.data, item=item)
+        db.session.add(discount_price)
+        db.session.commit()
+    return make_response(jsonify(status='ok', result={'message': 'New discount policy created', 'discount_policy': jsonify_discount_policy(discount_policy)}), 201)
+
+
+@app.route('/admin/discount_policy/<discount_policy_id>/edit', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_edit_discount_policy(discount_policy):
+    discount_policy_form = DiscountForm()
+    discount_policy_form.items.query = Item.query.all()
+    if not discount_policy_form.validate_on_submit():
+        return make_response(jsonify(status='error', error='invalid_details', error_description=discount_policy_form.errors), 400)
+    if not discount_policy_form.discount_type.data:
+        discount_policy_form.discount_code_base.data = None
+    discount_policy_form.populate_obj(discount_policy)
+    items = request.form.getlist('items')
+    for item_id in items:
+        item = Item.query.get(item_id)
```

Comment:
Given a set of `item_ids` you can fetch all the items at once, instead of running the query in a loop.